server/admin_management/api/app/services/access_control_privilege_service.py

```diff
@@ -0,0 +1,150 @@
+        LOGGER.debug(
+            f"Request for assigning access role privilege to a user: {request}."
+        )
+
+        # Verify if user already exists or add a new user
+        fam_user = self.user_service.find_or_create(
+            request.user_type_code, request.user_name, requester
+        )
+
+        # Verify if role exists
+        fam_role = self.role_service.get_role_by_id(request.role_id)
+        if not fam_role:
+            error_msg = f"Role id {request.role_id} does not exist."
+            utils.raise_http_exception(HTTPStatus.BAD_REQUEST, error_msg)
+
+        # Role is a 'Abstract' type, create access control privilege with forst client child role
+        require_child_role = (
+            fam_role.role_type_code == famConstants.RoleType.ROLE_TYPE_ABSTRACT
+        )
+
+        create_return_list: List[schemas.FamAccessControlPrivilegeCreateResponse] = []
+
+        if require_child_role:
+            LOGGER.debug(
+                f"Role {fam_role.role_name} requires child role "
+                "for creating delegate admin access control privilege."
+            )
+
+            if (
+                not hasattr(request, "forest_client_number")
+                or request.forest_client_number is None
+            ):
+                error_msg = (
+                    "Invalid access control privilege request, missing forest client number."
+                )
+                utils.raise_http_exception(HTTPStatus.BAD_REQUEST, error_msg)
+
+            for forest_number in request.forest_client_number:
+                # Check if child role exists or add a new child role
+                child_role = self.role_service.find_or_create_forest_client_child_role(
+                    forest_number, fam_role, requester
+                )
+                handle_create_return = self.handle_access_control_privilege_create(
+                    fam_user.user_id, child_role.role_id, requester
+                )
+                create_return_list.append(handle_create_return)
+
+        else:
+            handle_create_return = self.handle_access_control_privilege_create(
+                fam_user.user_id, fam_role.role_id, requester
+            )
+            create_return_list.append(handle_create_return)
+
+        LOGGER.debug(
+            f"Creating access control privilege executed successfully: {create_return_list}"
+        )
+
+        return create_return_list
+
+    def handle_access_control_privilege_create(
+        self, user_id: int, role_id: int, requester: str
+    ) -> schemas.FamAccessControlPrivilegeCreateResponse:
+        access_control_privilege_return = None
+
+        # Check if user privilege already exists
+        fam_access_control_privilege = self.get_acp_by_user_id_and_role_id(
+            user_id, role_id
+        )
+
+        if fam_access_control_privilege:
+            error_msg = "User already has the requested access control privilege."
```

Comment:
This might not be clear if message is intended for frontend to display.

Reply:
Updated to be like ` "error_message": "User already has the requested access control privilege for FOM_SUBMITTER_00001013"`